src/ParallelAlgorithms/Interpolation/Callbacks/ObserveTimeSeriesOnSurface.hpp

```diff
@@ -40,9 +40,35 @@ namespace callbacks {
 
 namespace detail {
 
+template<typename T>
+struct is_array_of_double : std::false_type {};
+
+template<std::size_t N>
+struct is_array_of_double<std::array<double, N>> : std::true_type {};
+
 template <typename... Ts>
 auto make_legend(tmpl::list<Ts...> /* meta */) {
-  return std::vector<std::string>{"Time", db::tag_name<Ts>()...};
+    std::vector<std::string> legend = {"Time"};
+
+    auto append_tags = [&legend](auto tag) {
+        using TagType = decltype(tag);
+        using ReturnType = typename TagType::type;
+
+        constexpr std::array<const char*, 3> pattern = {"_x", "_y", "_z"};
```

Comment:
GCC 9 wants you to move this line into the `if constexpr` (see the failed check)